pkgs/applications/audio/soundconverter/default.nix

```diff
@@ -0,0 +1,80 @@
+{ lib, fetchurl
+# Optional due to unfree license.
+, faacSupport ? false
+, python3Packages, gtk3, wrapGAppsHook
+, gsettings-desktop-schemas, intltool, xvfb-run
+, gobject-introspection, gst_all_1, fdk-aac-encoder }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.3";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "sha256-hzIG/4LD3705erPYvXb7uoRwF9LtKKIKB3jrhpYMsZ0=";
+  };
+
+  buildInputs = [
+    gtk3
+    wrapGAppsHook
+  ];
+
+  nativeBuildInputs = [
+    intltool
+    gobject-introspection
+  ];
+
+  propagatedBuildInputs = [
+    gobject-introspection
+    fdk-aac-encoder
+    python3Packages.gst-python
+    python3Packages.distutils_extra
+    python3Packages.setuptools
+    gst_all_1.gst-libav
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-ugly
+    (gst_all_1.gst-plugins-bad.override { inherit faacSupport; })
+  ];
+
+  checkInputs = [
+    xvfb-run
+  ];
+
+  preCheck = let
+    gds = gsettings-desktop-schemas;
```

Comment:
This does not really save that much.

Reply:
Yes it does. It makes the line 78 chars long and readable in a vertical split instead of 122 chars long and unreadable.

Not fixing that.

Reply:
In that case, it might be nicer to refactor to something like `lib.concatMapStringsSep ":" ("${pkg}/share/gsettings-schemas/${pkg.name}") [ gsettings-desktop-schemas gtk3 { outPath = "$out"; name = "${pname}-${version}"; } ]`

Reply:
Eeeee, does that really make more readable? Seems like that makes it less readable.

Reply:
At least with simple calls to `addToSearchPath` it's very clear what's happening, but when i see some kind of `concatMapStringsSep` it's not immediately clear what's happening. I don't really see the benefit.

Reply:
Even if there was a way to reference the current derivation, like `self`(hypothtically) it would look something like this:
```nix
{
  preCheck = let
    genXdgPath = pkg: "${pkg}/share/gsettings-schemas/${pkg.name}";
    combinePaths = paths: lib.concatMapStringsSep ":" genXdgPath paths;
  in ''
    export HOME=$TMPDIR
    addToSearchPath XDG_DATA_DIRS ${combinePaths [ gsettings-desktop-schemas gtk3 self ]}
  '' + lib.optionalString (!faacSupport) ''
    substituteInPlace tests/testcases/integration.py --replace \
      "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
      "for encoder in ['fdkaacenc', 'avenc_aac']:"
  '';
}
```
Which isn't really any more clear than the current:
```nix
{
  preCheck = let
    gds = gsettings-desktop-schemas;
  in ''
    export HOME=$TMPDIR
    addToSearchPath XDG_DATA_DIRS "${gds}/share/gsettings-schemas/${gds.name}"
    addToSearchPath XDG_DATA_DIRS "${gtk3}/share/gsettings-schemas/${gtk3.name}"
    addToSearchPath XDG_DATA_DIRS "$out/share/gsettings-schemas/${pname}-${version}"
  '' + lib.optionalString (!faacSupport) ''
    substituteInPlace tests/testcases/integration.py --replace \
      "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
      "for encoder in ['fdkaacenc', 'avenc_aac']:"
  '';
}
```
And we don't have a `self` to use, so we have to use `{ outPath = "$out"; name = "${pname}-${version}"; }`, which just doesn't look good, and as we both know programming is all about aesthetics.

Reply:
I do indeed think the first example is nicer (and will be even more so with https://github.com/NixOS/nixpkgs/pull/153246). But in the end it is your call as you are the maintainer.

Reply:
And I would disagree. Just because something repeats(twice, really) does not mean turning it into an expression involving lists, concatenation, separators, and helper functions, makes it any more readable or simpler.

Now, with your addition of `getSchemaDataDirPath` we get less boilerplate, so that's nice, but I do not think it's more readable. If we were doing more calls to `addToSearchPath` then maybe, but it just seems like premature optimization for future expansion of something that will probably not be touched for years. I'll be happy to use `getSchemaDataDirPath`, but I do think the addition of `concatMapStringsSep` and that weird `self` thing - which takes a moment to figure out why that's necessary in that form - does not help readability.

But I don't see the point of arguing about it any further, so I'll just do that once your PR is merged.